tests/python/test_tensor_consistency.py

```diff
@@ -0,0 +1,323 @@
+import numpy as np
+import pytest
+
+import gstaichi as ti
+
+from tests import test_utils
+
+_TI_DTYPE_TO_NP_DTYPE = {
+    ti.u1: np.bool_,
+    ti.u8: np.uint8,
+    ti.u16: np.uint16,
+    ti.u32: np.uint32,
+    ti.u64: np.uint64,
+    ti.i8: np.int8,
+    ti.i16: np.int16,
+    ti.i32: np.int32,
+    ti.i64: np.int64,
+}
+
+
+@pytest.mark.parametrize("tensor_type", [ti.field, ti.ndarray])
+@pytest.mark.parametrize("dtype", [ti.u1, ti.u8, ti.u16, ti.u32, ti.u64, ti.i8, ti.i32, ti.i16, ti.i64])
+@test_utils.test()
+def test_tensor_consistency_kernel_write_to_numpy_consistency(tensor_type, dtype) -> None:
+    """
+    write from kernel => to_numpy => check
+    """
+    assert ti.cfg is not None
+    arch = ti.cfg.arch
+    if dtype == ti.u1 and arch in [ti.vulkan, ti.metal]:
+        pytest.xfail("u1 doesnt work on vulkan or metal doesn't work currently, neither on field nor ndarray")
+
+    poses = [0, 2, 5, 11]
+    a = tensor_type(dtype, (16,))
+
+    TensorType = ti.types.NDArray if tensor_type == ti.ndarray else ti.Template
+
+    @ti.kernel
+    def k1(a: TensorType) -> None:
+        for b_ in range(1):
+            for pos in ti.static(poses):
+                a[pos] = 1
+
+    k1(a)
+
+    a_np = a.to_numpy()
+
```

Comment:
I would avoid spreading this in too many lines, but as you want.

Reply:
yeah, I prefer to separate out the different concepts, ie
- kernel call
- copy to host
- check